v2v/tests/src/libnbd/libnbd.py

```
@@ -95,6 +100,91 @@ def test_unsanitized_hostname():
+        3. Create a 100M destination image fully allocated with zeros (non-sparse).
+        4. Verify destination initial disk usage is 100M.
+        5. Run 'nbdcopy' with --destination-is-zero from source to destination.
+        6. Verify destination disk usage has decreased to match source sparseness (~4K).
+        7. Perform a binary comparison (cmp) to ensure data integrity.
+        """
+        # Configuration
+        SRC_IMG = os.path.join(data_dir.get_tmp_dir(), "src.img")
+        DEST_ZERO = os.path.join(data_dir.get_tmp_dir(), "dest_zero.img")
+
+        # 1. Create source with small data (sparse)
+        process.run(f"truncate -s 100M {SRC_IMG}", shell=True, ignore_status=True)
```

Ruff (0.15.9)

[error] Function call with `shell=True` parameter identified, security issue

(S604)